spec/controllers/request_controller_spec.rb

```diff
@@ -604,9 +604,9 @@ def get_attachment(params = {})
                            :part => 2,
                            :file_name => 'interesting.html',
                            :skip_cache => 1
-      expect(response.body).not_to match("script")
-      expect(response.body).not_to match("interesting")
-      expect(response.body).to match('dull')
+      expected =
+        "<html>\n  <head>\n  </head>\n  <body>dull\n    \n  </body>\n</html>\n"
+      expect(response.body).to eq(expected)
```

Comment:
First, some examples:

```ruby
content = <<-EOF
<html>
  <head>
  </head>
  <body>dull
    <script>alert('interesting')</script>
  </body>
</html>
EOF

# Rails 4.1
helper.sanitize(content)
# => "<html>\n  <head>\n  </head>\n  <body>dull\n    \n  </body>\n</html>\n"

# Rails 4.2
#
# Default sanitize method with custom set of
# config.action_view.sanitized_allowed_tags
helper.sanitize(content)
# => "\n  \n  \n  dull\n    alert('interesting')\n  \n\n"

# Liz's new helper:
helper.sanitize_html(content)
# => "\n  \n  \n  dull\n    \n  \n\n"

# Default with a supplied list of allowed tags.
# For some reason these don't work as I'd expect
helper.sanitize(content, tags: %w(html head body))
# => "\n  \n  \n  dull\n    alert('interesting')\n  \n\n"
helper.sanitize(content, tags: %w(body))
# => "\n  \n  \n  dull\n    alert('interesting')\n  \n\n"

# Try _allowing_ the script tag. Works, but still strips out the other tags.
helper.sanitize(content, tags: %w(script))
# => "\n  \n  \n  dull\n    <script>alert('interesting')</script>\n  \n\n"
```

What's going on?

We're [parsing](https://github.com/rails/rails-html-sanitizer/blob/master/lib/rails/html/sanitizer.rb#L33) `content` as a Loofah **fragment**. Since `content` is a full HTML document, we [should](https://github.com/flavorjones/loofah#side-note-fragments-vs-documents) be parsing it as a Loofah **document**.

I think we _do_ need a custom sanitzer helper for sanitizing attachments (as per ed87595fc52d0aa020b32e124fdbfd4bbe672e84), but just adding an additional scrubbing step is incorrect.

I think we should define a new sanitzer in the same way as [`Rails::Html::FullSanitizer`](https://github.com/rails/rails-html-sanitizer/blob/master/lib/rails/html/sanitizer.rb#L22-L40), replacing the Loofer line with `Loofer.document`.